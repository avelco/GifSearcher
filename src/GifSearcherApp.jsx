import { useState } from "react"
import { AddCategory, GifGrid } from "./components"

export const GifSearcherApp = () => {

  const [categories, setCategories] = useState(['One Punch'])

  const onAddCategory = (newCategory) => {

    if ( categories.includes(newCategory) ) return;

    setCategories( [newCategory, ...categories, ] )
    
  }

  return (
    <>

        <h2>GifSearcherApp</h2>

        <AddCategory 
          // setCategories={ setCategories }
          onNewCategory={ onAddCategory }
        />

		{ 
			categories.map( (category) => {
				return <GifGrid 
				key={ category }
				category={ category }/>
			}) 
		}
    </>
  )
}
