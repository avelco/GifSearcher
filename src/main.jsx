import React from 'react';
import ReactDOM from 'react-dom/client';
import { GifSearcherApp } from './GifSearcherApp.jsx';

import './styles.css';  // Import CSS

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    <GifSearcherApp />
  </React.StrictMode>,
)
