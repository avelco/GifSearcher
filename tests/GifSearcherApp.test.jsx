import { render, screen } from "@testing-library/react";
import { GifSearcherApp } from "../src/GifSearcherApp";

describe('GifSearcherApp', () => {

    test('should render GifSearcherApp component and be equal to the snapshot', () => {

        const wrapper = render( <GifSearcherApp /> );
        expect( wrapper ).toMatchSnapshot();

    });

    test('should render a title', () => {

        render( <GifSearcherApp /> );
        expect( screen.getByText('GifSearcherApp') ).toBeTruthy();
    });

    test()
});