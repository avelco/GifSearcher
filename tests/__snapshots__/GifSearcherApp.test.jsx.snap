// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifSearcherApp should render GifSearcherApp component and be equal to the snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <h2>
        GifSearcherApp
      </h2>
      <form
        aria-label="form"
      >
        <input
          type="text"
          value="Rick and Morty"
        />
      </form>
      <h3>
        One Punch
      </h3>
      <p>
        Loading...
      </p>
      <div
        class="card-grid"
      />
    </div>
  </body>,
  "container": <div>
    <h2>
      GifSearcherApp
    </h2>
    <form
      aria-label="form"
    >
      <input
        type="text"
        value="Rick and Morty"
      />
    </form>
    <h3>
      One Punch
    </h3>
    <p>
      Loading...
    </p>
    <div
      class="card-grid"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
